src/sql/workbench/services/insights/browser/insightsDialogView.ts

```diff
@@ -337,7 +337,11 @@ export class InsightsDialogView extends Modal {
 				let task = tasks.includes(action);
 				let commandAction = MenuRegistry.getCommand(action);
 				let commandLabel = types.isString(commandAction.title) ? commandAction.title : commandAction.title.value;
-				if (task && !this.findFooterButton(commandLabel)) {
+				if (task) {
+					// need to remove and add fresh because the onDidClick action from previous addition is not called
+					if (this.findFooterButton(commandLabel)) {
```

Comment:
Oh there's already this method for finding the button. I'd suggest either having removeFooterButton do the check as well and just take in the button object directly or remove the findFooterButton call here and just add it to removeFooterButton

Reply:
This increases readability (to get a sense of what is happening exactly) so want to keep it this way

Reply:
Ok, but you shouldn't be duplicating the find logic. Feel free to keep it split into two parts but this is just making it take longer than it needs to.

Reply:
hmm.... I don't necessarily worry with this less cost. But I am fine with changing to single call too... will remove the find call as you suggested :)